src/app/[locale]/incident/add/components/IncidentQuestionsLocationForm.tsx

```diff
@@ -92,12 +100,31 @@ export const IncidentQuestionsLocationForm = () => {
       extra_properties: answers,
     })
 
-    setLastCompletedStep(2)
+    addVisitedStep(FormStep.STEP_2_ADD)
     addOneStep()
-
-    router.push('/incident/contact')
+    router.push(steps[FormStep.STEP_3_CONTACT])
   }
 
+  useEffect(() => {
+    if (navToSummary) {
+      // todo: finish update form when component it fixed
+      // updateForm({})
+
+      goToStep(FormStep.STEP_4_SUMMARY)
+      router.push(steps[FormStep.STEP_4_SUMMARY])
+      onNavToSummary(false)
+    }
+  }, [navToSummary])
+
+  useEffect(() => {
+    if (goBack) {
+      // updateForm
+      goToStep(FormStep.STEP_1_DESCRIPTION)
+      router.push(steps[FormStep.STEP_1_DESCRIPTION])
+      onGoBack(false)
+    }
+  }, [goBack])
```

Comment:
Wat denk ik een handigere manier is om al deze code duplicatie in straks elke form te voorkomen is deze goBack useEffect logica en summary logica in het FormProgress component te stoppen.

Wat je in principe namelijk wil is dat bij een klik op vorige dan een functie gecalled wordt, die obv van dan je huidige formulier stap de nieuwe stap bedenkt: goToStep(betreffendeStap) daarmee zet, dan de router.push doet en onGoBack(false) zet, dan weer datzelfde riedeltje voor de summary.

Form specifieke logica (om het formulier op te slaan) hou je dan gewoon in de form submission functie en die call je dan ook weer vanuit die functie in je form progress component.